### DISCUSSION TITLE
creating components from API

### DISCUSSION THREAD

**Discussion by endquote** [WeblateOrg/weblate#9428] [General]
I'm trying to use [the API to create components](https://docs.weblate.org/en/weblate-4.17/api.html#post--api-projects-(string-project)-components-). An example call:

```ts
await weblate.post(`/projects/${project.id}/components/`, {
  name: project.name,
  slug: project.id,
  repo: gitRepo,
  filemask: "*.xml",
  file_format: "xliff",
  branch: project.id,
  template: "base.xml",
  new_base: "base.xml",
});
```

I created a repo and made sure weblate could access it by ssh. I made the branch, and added `base.xml` file like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
</xliff>
```

I get this error after making the call:

`Could not parse translation base file: File parse error.`

I could try it without the files:

```ts
await weblate.post(`/projects/${project.id}/components/`, {
  name: project.name,
  slug: project.id,
  repo: gitRepo,
  filemask: "*.xml",
  file_format: "xliff",
  branch: project.id,
});
```

Then I get:

`You have set up Weblate to add new translation files, but did not provide a base file to do that.`

I could try even less:

```ts
await weblate.post(`/projects/${project.id}/components/`, {
  name: project.name,
  slug: project.id,
});
```
I get an error saying that `repo`, `filemask`, and `file_format` is required.

I do not actually have any translations to import, I plan to post them via the API. Something like the "Start from scratch" option in the web UI would work for me, but I'm not sure how to do that from the API.

**Comment by endquote** [comment]
I'm going to try a different approach — exporting from my current tool to a format Weblate can import, and then putting that in git.

**Comment by nijel** [comment]
> I get this error after making the call:
> 
> `Could not parse translation base file: File parse error.`

There should be a more detailed error message here. Can you check server logs, what went wrong?
